catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/metacard/UnionSingleMetacardUpdater.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.video.stream.mpegts.metacard;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+
+/**
+ * Set the parent to a single value of the union of the child and parent values.
+ */
+public class UnionSingleMetacardUpdater implements MetacardUpdater {
+
+    private static final int VALUE_INDEX = 0;
+
+    private final String attributeName;
+
+    public UnionSingleMetacardUpdater(String attributeName) {
+        this.attributeName = attributeName;
+    }
+
+    @Override
+    public final void update(Metacard parent, Metacard child) {
+
+        List<Serializable> serializables = Stream.of(parent, child)
+                .map(this::getAttribute)
+                .filter(Objects::nonNull)
+                .map(Attribute::getValues)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .distinct()
+                .collect(Collectors.toList());
```

Comment:
This is a weird class, as it unions the parent and child values and then, basically, pulls one of them at random from that union. If that's actually of value - which I'm having trouble visualizing - then it would be simpler to skip the `distinct` and `collect` calls and just `findFirst` after the `flatMap`, I think.